solvation_analysis/plotting.py

```diff
@@ -0,0 +1,319 @@
+    # ideally this same API could be reused throughout the plotting API
+
+    speciation_data = solution.speciation.speciation_data
+    speciation_data["total"] = speciation_data.sum(axis=1)
+    sums = speciation_data.groupby("total").sum()
+    fig = go.Figure()
+    totals = sums.T.sum()
+    for column in sums.columns:
+        fig.add_trace(go.Bar(x=sums.index.values, y=sums[column].values/totals, name=column))
+    fig.update_layout(xaxis_title_text="Shell Size", yaxis_title_text="Fraction of Total Molecules",
+                      title="Fraction of Solvents in Shells of Different Sizes")
+    fig.update_xaxes(type="category")
+    return fig
+
+
+def plot_speciation(solution):
+    # square area
+    # should be doable with plotly.express.imshow and go.add_annotations
+    return
+
+
+def plot_co_occurrence(solution):
+    return
+
+
+def plot_clustering(solution):
+    # not in this branch yet
+    return
+
+
+def plot_coordinating_atoms(solution):
+    # for each solvent
+    # by atom type? could allow by element or other features?
+    # bar chart with one bar for each solvent
+    # normalized
+    return
+
+
+# multiple solutions
+def compare_solvent_dicts(properties, coerce_solvent_names, keep_solvents, legend_label, x_axis="species", series=False):
+    # generalist plotter, this can plot either bar or line charts of the same data
+    """
+
+    Parameters
+    ----------
+    properties : dictionary of the solvent property to be compared
+    coerce_solvent_names : a dictionary where the keys are strings of solvent names and the values are
+        strings of a more generic name for the solvent (i.e. {"EAf" : "EAx", "fEAf" : "EAx"})
+    keep_solvents : a list of strings of solvent names that are common to all systems in question,
+        graphed in the order that is passed into the function
+    legend_label : title of legend as a string
+    x_axis : a string specifying "species" or "solution" to be graphed on the x_axis
+    series : Boolean (False for a bar graph; True for a line graph)
+
+    Returns
+    -------
+    fig : Plotly.Figure (generic plot)
+
+    """
+    # coerce solutions to a common name
+    for solution in coerce_solvent_names:
+        if solution in properties:
+            properties[solution][coerce_solvent_names[solution]] = properties[solution].pop(solution)
+
+    # filter out components of solution to only include those in keep_solvents
+    if keep_solvents:
+        for solution in properties:
+            try:
+                properties[solution] = {keep: properties[solution][keep] for keep in keep_solvents}
+            except KeyError:
+                # if argument for keep_solvents is invalid
+                raise Exception("Invalid value of keep_solvents. \n keep_solvents: " +
+                                str(keep_solvents) + "\n Valid options for keep_solvents: " +
+                                str(list(properties[solution].keys()))) from None
```

Comment:
If possible, let's try to avoid the `try except` loop.

Look into `Dict.get()` which returns the item if it exists and returns `None` if not. It's a very convenient function!